<launch>
    <!-- Relevant args that might be changed at launch time -->
    <arg name="input_cloud_topic"       default="/camera/depth/color/points" />
    <arg name="start_realsense"         default="false" />

    <!-- Camera-related arguments -->
    <arg name="camera_width"                            default="640" />
    <arg name="camera_height"                           default="480" />
    <arg name="camera_fps"                              default="15" />

    <!-- Start realsense, if necessary -->
    <include file="$(find realsense2_camera)/launch/rs_rgbd.launch" if="$(arg start_realsense)">
        <arg name="align_depth"                         value="true" />
        <arg name="depth_registered_processing"         value="true" />
        <arg name="enable_pointcloud"                   value="true" />
        <arg name="filters"                             value="hole_filling" /> <!-- Change this if need be -->
        <arg name="depth_width"                         value="$(arg camera_width)" />
        <arg name="depth_height"                        value="$(arg camera_height)" />
        <arg name="depth_fps"                           value="$(arg camera_fps)" />
        <arg name="color_width"                         value="$(arg camera_width)" />
        <arg name="color_height"                        value="$(arg camera_height)" />
        <arg name="color_fps"                           value="$(arg camera_fps)" />
    </include>

    <!-- Start the actual segmentation node and remap topics -->
    <rosparam command="load" file="$(find tabletop_segment)/cfg/tabletop_segment.yaml" ns="tabletop_segment" />
    <remap from="/tabletop_segment/input_cloud" to="$(arg input_cloud_topic)" />
    <node name="tabletop_segment" pkg="tabletop_segment" type="tabletop_segment_node" />

</launch>